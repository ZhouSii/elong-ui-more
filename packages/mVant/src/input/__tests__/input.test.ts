import { expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import EvInput from '../index';

test('EvInput placeholder', () => {
  const wrapper = mount(EvInput, {
    props: {
      placeholder: '12323232'
    }
  });

  expect(wrapper.html()).toMatchInlineSnapshot(
    '"<van-field modelvalue=\\"\\" placeholder=\\"12323232\\" label-width=\\"90\\" data-v-0ce8b691=\\"\\"></van-field>"'
  );
});

test('EvInput isInput slots', () => {
  const wrapper = mount(EvInput, {
    props: {
      isInput: false
    },
    slots: 'Custom Input'
  });

  expect(wrapper.text()).toMatchInlineSnapshot('""'); // 替换成你期望的默认内容
});

test('should render word limit correctly when modelValue is undefined', () => {
  const wrapper = mount(EvInput, {
    props: {
      modelValue: undefined,
      maxlength: 3,
      showWordLimit: true
    }
  });
  expect(wrapper.html()).toMatchInlineSnapshot(
    '"<van-field maxlength=\\"3\\" showwordlimit=\\"true\\" modelvalue=\\"\\" placeholder=\\"请输入\\" label-width=\\"90\\" data-v-0ce8b691=\\"\\"></van-field>"'
  );
});
