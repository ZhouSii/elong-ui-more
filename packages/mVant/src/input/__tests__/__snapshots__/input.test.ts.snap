// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render word limit correctly when modelValue is undefined 1`] = `"<van-field maxlength=\\"3\\" showwordlimit=\\"true\\" modelvalue=\\"\\" placeholder=\\"请输入\\" label-width=\\"90\\" data-v-0ce8b691=\\"\\"></van-field>"`;
